hamilton/function_modifiers/expanders.py

```diff
@@ -454,14 +454,28 @@ def expand_node(
         fn = node_.callable
         base_doc = node_.documentation
 
-        def df_generator(*args, **kwargs) -> Any:
-            df_generated = fn(*args, **kwargs)
-            if self.fill_with is not None:
-                for col in self.columns:
-                    if col not in df_generated:
-                        registry.fill_with_scalar(df_generated, col, self.fill_with)
-                        assert col in df_generated
-            return df_generated
+        # if fn is an async function
+        if inspect.iscoroutinefunction(fn):
+
+            async def df_generator(*args, **kwargs) -> Any:
+                df_generated = await fn(*args, **kwargs)
+                if self.fill_with is not None:
```

Comment:
This should use the same code as the next one

Reply:
We should have a utility function that it calls to

Reply:
see commit message.